package ru.yandex.practicum.sleeptracker;

import org.junit.jupiter.api.Test;

import java.time.LocalDateTime;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

class MinDurationFunctionTest {

    @Test
    void minDuration_shouldReturnShortestSession() {

        List<SleepingSession> sessions = List.of(
                new SleepingSession(
                        LocalDateTime.of(2025, 10, 1, 22, 15),
                        LocalDateTime.of(2025, 10, 2, 8, 0),
                        SleepQuality.GOOD),
                new SleepingSession(
                        LocalDateTime.of(2025, 10, 2, 23, 0),
                        LocalDateTime.of(2025, 10, 3, 8, 0),
                        SleepQuality.NORMAL),
                new SleepingSession(
                        LocalDateTime.of(2025, 10, 3, 14, 30),
                        LocalDateTime.of(2025, 10, 3, 15, 20),
                        SleepQuality.NORMAL),
                new SleepingSession(
                        LocalDateTime.of(2025, 10, 3, 23, 30),
                        LocalDateTime.of(2025, 10, 4, 6, 20),
                        SleepQuality.BAD)
        );

        MinDurationFunction function = new MinDurationFunction();
        SleepAnalysisResult result = function.apply(sessions);

        assertEquals("50", result.getValue());
    }

    @Test
    void minDuration_shouldReturnZeroForEmptyList() {

        List<SleepingSession> sessions = List.of();

        MinDurationFunction function = new MinDurationFunction();
        SleepAnalysisResult result = function.apply(sessions);

        assertEquals("0", result.getValue());
    }
}
